soroban-sdk-macros/src/lib.rs

```diff
@@ -95,6 +97,45 @@ pub fn contractimpl(_metadata: TokenStream, input: TokenStream) -> TokenStream {
     }
 }
 
+#[derive(Debug, FromMeta)]
+struct MetadataArgs {
+    key: String,
+    val: String,
+}
+
+#[proc_macro]
+pub fn metadata(metadata: TokenStream) -> TokenStream {
+    let args = parse_macro_input!(metadata as AttributeArgs);
+    let args = match MetadataArgs::from_list(&args) {
+        Ok(v) => v,
+        Err(e) => return e.write_errors().into(),
+    };
+
+    let gen = {
+        //TODO: is key necessary?
+        let meta_v0 = ScMetaV0 {
+            key: args.key.clone().try_into().unwrap(),
+            val: args.val.try_into().unwrap(),
```

Comment:
Ideally these `.unwrap()` calls are replaced with errors that get rendered with the compiler pointing at the respective code item and saying what's wrong with it. Take a look at how errors are generated for some of the other macros using the `syn::Error` type and converting it into a `TokenStream`:

https://github.com/stellar/rs-soroban-sdk/blob/cd01ad4327095132ef00c9f389d29f3b9f589e85/soroban-sdk-macros/src/lib.rs#L229-L231

Reply:
done